packages/api-docs/src/PageJsonGenerator.tsx

```diff
@@ -182,98 +194,141 @@ function extractText(node: DocNode): string {
 }
 
 /**
- * Creates the interface page json object
- *
- * @param collectedData - Collected data to use for linking
- * @param interfaceItem - Interface item to search
+ * Render a link into text. For now we just extract the text or the code item name
+ * (rather than returning an actual link).
  */
-function createInterfacePageJson(collectedData: CollectedData, interfaceItem: ApiInterface): ITableJson {
-  const interfaceTableRowJson: ITableRowJson[] = [];
+function renderLinkTag(apiModel: ApiModel, apiItem: ApiItem, link: DocLinkTag): string {
+  if (link.linkText) {
+    return link.linkText;
+  } else if (link.codeDestination) {
+    const result = apiModel.resolveDeclarationReference(link.codeDestination, apiItem);
+    if (result.resolvedApiItem) {
+      return result.resolvedApiItem.getScopedNameWithinPackage();
+    }
+  }
+  return '';
+}
 
+/**
+ * Generate an ITableJson for any top-level API item (interface, class, enum, type alias)
+ * with the name, description, deprecated message, and optionally extends tokens pre-filled.
+ */
+function getTableJson(
```

Comment:
This function (and a few more in this file that are similar) are fine, but they contain enough logic that I wonder if `ITableJson` and `ITableRowJson` would be better implemented as full-on classes instead of interfaces. Then getters could be defined for things like `deprecated` and the logic for that bit would be obvious.

Reply:
Interesting idea. I'm not going to change it now but I'll keep that in mind.